### ISSUE TITLE
Support the mongoDB change stream API

### ISSUE THREAD

**Issue by maelle-le-bon** [typeorm/typeorm#1740]
**Issue type:**

[ ] question
[ ] bug report
[ X ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ X ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `websql`

**TypeORM version:**

[ X ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Demo with the native mongo driver:**
Since mongo 3.6, the change stream API allow to watch changes on a collection.
```ts
let mongoUrl: string = 'mongodb://192.168.1.3:27017';
let client: MongoClient = await MongoClient.connect(mongoUrl);
let db: Db = client.db('nsrp');
let accountsCollection: Collection = db.collection('accounts');

accountsCollection.watch({ fullDocument: 'updateLookup' }).on('change', (change) => {
	switch (change.operationType) {
		case 'update':
			console.log('Someone change: ' + change.fullDocument.username);
			break;

		case 'insert':
			console.log('New user created: ' + change.fullDocument.username);
			break;

		case 'delete':
			console.log('A user were deleted: ' + change.documentKey._id);
			break;
	}
});
```

**Possible implementations in typeorm:**
I'm new in typeorm so I'm not very familiar with the architecture but this is what I think would be the best:

- Exposing the mongo collection object could offer all futur features that the mongo driver offer before abstracting them on the typeorm ORM.
- A watch() method could be added on the MongoRepository class or/and in the BaseEntity class

I will explore the typeorm structur to find how I can add this feature but I think it could be a big advantage to offer a simple way to watch changes on the collection.

**Comment by adam-lebon** [comment]
Just seen that typeorm doesn't support mongodb driver >=3.0 and the watch api isn't available at 2.2.
So this feature will not be available since typeorm does not support mongo 3 driver.

**Comment by pleerock** [comment]
There is a PR for mongo 3 support, but author is still working on it.

This feature is opened for contributions, looks like simply need to add this method into MongoRepository and MongoEntityManager classes.

**Comment by trubit** [comment]
Looks like Mongo driver v 3 is already supported. This issue can now be fixed.

**Comment by rustamwin** [comment]
:+1:

**Comment by rustamwin** [comment]
This issue fixed #3699
